### PR TITLE
user preferences are stored separately and merged with organization rules

### LINKED ISSUE #512: sync personal rules to the API
<p></p>

### LINKED ISSUE #528: return a 403 unless user is authenticated

### PR DESCRIPTION
**Pull request by lsmith77**
required for https://github.com/witty-works/dashboard/issues/164

fixes https://github.com/witty-works/nlp_api/issues/512
fixes https://github.com/witty-works/nlp_api/issues/528
fixes https://github.com/witty-works/browser-extension/issues/499

Note, due to BC breaks there is now a `/v2.0/check` and `/v2.0/auth` endpoints and the `/check` endpoint has been dropped

### CODE REVIEW

File: tests/test_auth_2_0/test_auth/output.json [hunk 1]
```diff
@@ -0,0 +1,34 @@
+{
+    "config": {
```
Comment (lsmith77) [inline tests/test_auth_2_0/test_auth/output.json:2]: add domain examples

File: app/models.py [hunk 2]
```diff
@@ -506,42 +497,45 @@ def factory(
...
+            "icon": icon,
+            "url": url,
+            "context": explanation_context,
+        }
+
+        if version <= 1.1:
+            if gravity != None:
+                gravity = math.ceil(gravity)
+
             return ResultOutOld(
```
Comment (lsmith77) [inline app/models.py]: ResultOutOld => ResultOut1_1

File: app/main.py [hunk 3]
```diff
@@ -377,27 +418,18 @@ async def check_v1_1(
         results=results,
         language=language,
         limit_reached=limit_reached,
-        organization_config=organization_config,
+        config_changed=get_config_change(rules, user_request_in),
     )
 
 
 # data exchange routes
 @app.post("/store_rules")
```
Comment (lsmith77) [inline app/main.py:430]:
```
for user in organization_rules.users:
```

**Comment by lsmith77** [comment]
@hellasol ok .. I made all the tweaks we discussed today. so please review. we will then merge it and do some testing with the dev dashboard and browser extension to see if the BC layer does it's job.

**Review by hellasol — APPROVED** [review]

**Pull request merged**